datafusion/expr/src/expr_rewriter/mod.rs

```diff
@@ -181,10 +178,14 @@ pub fn create_col_from_scalar_expr(
             Some::<TableReference>(subqry_alias.into()),
             name,
         )),
-        Expr::Column(Column { relation: _, name }) => Ok(Column::new(
-            Some::<TableReference>(subqry_alias.into()),
+        Expr::Column(Column {
+            relation: _,
             name,
-        )),
+            spans,
+        }) => Ok(
+            Column::new(Some::<TableReference>(subqry_alias.into()), name)
+                .with_spans(spans.clone()),
+        ),
```

Comment:
You might be able to make this look nicer (and less likely to miss the span) via something like

```rust
        Expr::Column(col) => col.with_qualifier(subqry_alias.into())
```

Reply:
Good idea!